Packages/com.chisel.components/Chisel/Components/API.public/BrushMeshAssetFactory/Sphere/BrushFactory.Sphere.cs

```diff
@@ -15,40 +15,48 @@
 namespace Chisel.Components
 {
     public sealed partial class BrushMeshAssetFactory
-	{
-		public static bool GenerateSphereAsset(CSGBrushMeshAsset brushMeshAsset, CSGSphereDefinition definition)
-		{
-			var subMesh = new CSGBrushSubMesh();
-			if (!GenerateSphereSubMesh(subMesh, definition))
-			{
-				brushMeshAsset.Clear();
-				return false;
-			}
+    {
+        public static bool GenerateSphereAsset(CSGBrushMeshAsset brushMeshAsset, CSGSphereDefinition definition)
+        {
+            var subMesh = new CSGBrushSubMesh();
+            if (!GenerateSphereSubMesh(subMesh, definition))
+            {
+                brushMeshAsset.Clear();
+                return false;
+            }
 
-			brushMeshAsset.SubMeshes = new[] { subMesh };
-			brushMeshAsset.CalculatePlanes();
-			brushMeshAsset.SetDirty();
-			return true;
-		}
+            brushMeshAsset.SubMeshes = new[] { subMesh };
+            brushMeshAsset.CalculatePlanes();
+            brushMeshAsset.SetDirty();
+            return true;
+        }
 
-		public static bool GenerateSphereSubMesh(CSGBrushSubMesh subMesh, CSGSphereDefinition definition)
-		{
-			definition.Validate();
-			var transform = Matrix4x4.TRS(Vector3.zero, Quaternion.AngleAxis(definition.rotation, Vector3.up), Vector3.one);
-			return GenerateSphereSubMesh(subMesh, definition.diameterXYZ, transform, definition.horizontalSegments, definition.verticalSegments, definition.surfaceAssets, definition.surfaceDescriptions);
-		}
+        public static bool GenerateSphereSubMesh(CSGBrushSubMesh subMesh, CSGSphereDefinition definition)
+        {
+            definition.Validate();
+            var transform = Matrix4x4.TRS(Vector3.zero, Quaternion.AngleAxis(definition.rotation, Vector3.up), Vector3.one);
+            return GenerateSphereSubMesh(subMesh, definition.diameterXYZ, definition.generateFromCenter, transform, definition.horizontalSegments, definition.verticalSegments, definition.surfaceAssets, definition.surfaceDescriptions);
```

Comment:
Added option to be able to generate a sphere from its center, or from the bottom. Eventually this option needs to be exposed to the user so the user can decide on details like this depending on their needs